File: src/lib/painDetector.ts
const PAIN_KEYWORDS = ['hurt','hurts','pain','painful','sore','tight','tightness',
  'uncomfortable','sharp','ache','burning','pulling','strain','strained','injured','injury','weird','off']
export const detectPain = (msg: string): string[] =>
  PAIN_KEYWORDS.filter(kw => msg.toLowerCase().includes(kw))
export const hasPainSignal = (msg: string): boolean => detectPain(msg).length > 0

File: android/app/src/main/kotlin/com/untrained/app/domain/PainDetector.kt
package com.untrained.app.domain

object PainDetector {
    private val KEYWORDS = listOf(
        "hurt", "hurts", "pain", "painful", "sore", "tight", "tightness",
        "uncomfortable", "sharp", "ache", "burning", "pulling",
        "strain", "strained", "injured", "injury", "weird", "off"
    )

    fun detect(text: String): Boolean = KEYWORDS.any { text.lowercase().contains(it) }
}
